custom_components/tuya_local/devices/multi_leaf_dy_107_thermostat.yaml

```diff
@@ -0,0 +1,196 @@
+name: Multi-leaf DY-107 thermostat
+products:
+  - id: gdusjavy8i4dpjlt
+    name: Multi-leaf DY-107 thermostat
+primary_entity:
+  entity: climate
+  translation_key: thermostat
+  dps:
+    - id: 1
+      name: hvac_mode
+      type: boolean
+      mapping:
+        - dps_val: true
+          value: heat
+        - dps_val: false
+          value: "off"
+    - id: 2
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: auto
+          value: program
+        - dps_val: manual
+          value: manual
+    - id: 16
+      type: integer
+      name: temperature
+      range:
+        min: 5
+        max: 90
+    - id: 19
+      type: integer
+      name: max_temperature
+    - id: 24
+      type: integer
+      name: current_temperature
+    - id: 26
+      optional: true
+      type: integer
+      name: min_temperature
+    - id: 36
+      type: string
+      name: hvac_action
+      mapping:
+        - dps_val: close
+          icon: "mdi:radiator-disabled"
+          constraint: hvac_mode
+          conditions:
+            - dps_val: true
+              value: idle
+            - dps_val: false
+              value: "off"
+        - dps_val: open
+          constraint: hvac_mode
+          conditions:
+            - dps_val: true
+              icon: "mdi:radiator"
+              value: heating
+            - dps_val: false
+              icon: "mdi:radiator-disabled"
+              value: "off"
+    - id: 45
+      name: fault_code
+      type: bitfield
+secondary_entities:
+  - entity: number
+    name: Calibration offset
+    category: config
+    icon: "mdi:thermometer"
+    dps:
+      - id: 27
+        type: integer
+        name: value
+        unit: C
+        range:
+          min: -9
+          max: 9
+  - entity: switch
+    name: Anti-frost
+    icon: "mdi:snowflake-melt"
+    category: config
+    dps:
+      - id: 10
+        type: boolean
+        name: switch
+  - entity: lock
+    name: Child lock
+    category: config
+    dps:
+      - id: 40
+        optional: true
+        type: boolean
+        name: lock
+        mapping:
+          - dps_val: true
+            icon: "mdi:hand-back-right-off"
+          - dps_val: false
+            icon: "mdi:hand-back-right"
+  - entity: select
+    name: Sensor selection
+    category: config
+    dps:
+      - id: 43
+        type: string
+        name: option
+        mapping:
+          - dps_val: in
+            icon: "mdi:gauge"
+            value: Internal
+          - dps_val: out
+            icon: "mdi:thermometer-chevron-down"
+            value: External
+          - dps_val: all
+            icon: "mdi:thermometer"
+            value: Both
+  - entity: binary_sensor
+    name: Error
+    category: diagnostic
+    class: problem
+    dps:
+      - id: 45
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 0
+            value: false
+          - value: true
+  - entity: number
+    name: High temperature limit

```

Comment:
Usually I would use "High temperature limit" for an alarm point. Since this is setting the maximum for the temperature adjustment, I would use "Maximum temperature" here.  Similarly for "Minimum temperature" below.